packages/vscode-extension/lib/runtime.js

```diff
@@ -25,15 +25,74 @@ global.__RNIDE_onDebuggerReady = function () {
 // debug adapter and avoid exposing as part of application logs
 console.log("__RNIDE_INTERNAL", "radon-ide runtime loaded");
 
+let stackOffset = 1; // default offset is 1, because the first frame is the wrapConsole function
+let logRef = null;
+let errorRef = null;
+let warnRef = null;
+let infoRef = null;
+
+function isConsoleRefChanged() {
+  return (
+    logRef !== console.log ||
+    errorRef !== console.error ||
+    warnRef !== console.warn ||
+    infoRef !== console.info
+  );
+}
+
+function updateConsoleRefs() {
+  logRef = console.log;
+  warnRef = console.warn;
+  errorRef = console.error;
+  infoRef = console.info;
+}
+
 function wrapConsole(consoleFunc) {
+  let initializationStack = [];
+  let sendInitialLog = false;
+
   return function (...args) {
     const stack = parseErrorStack(new Error().stack);
-    const expoLogIndex = stack.findIndex((frame) => frame.methodName === "__expoConsoleLog");
-    const location = expoLogIndex > 0 ? stack[expoLogIndex + 1] : stack[1];
+
+    // To get the proper stack frame, so we can display link to the source code
+    // we need to skip wrappers (like wrapConsole below or for example Sentry wrapper)
+    // Otherwise, the stack frame would point to the wrapper and not the actual source code
+    // To do that, we run console.log again in runWrapper, and then compare
+    // first frames to find the offset. We do that when ant of console ref changes
+    if (isConsoleRefChanged()) {
```

Comment:
would be better to only check the current console function and not all functions. For that we should change `wrapConsole` to take `logFunctionKey` instead of actual function. It can also assign the wrapped function directly to console object so that we can just call `wrapConsole('log')`

Reply:
I think we need to take both. Otherwise, we lose ref to actual log method

Reply:
Hmm, I could skip the log function by saving it at the beginning, but the source translations stop working, let me check that.

Reply:
Ok, seems to be ok. Changed in https://github.com/software-mansion/radon-ide/pull/785/commits/bdce780109b07255849dc8bfc18718d581804e3c